xarray/core/dataset.py

```diff
@@ -2741,6 +2741,56 @@ def roll(self, **shifts):
 
         return self._replace_vars_and_dims(variables)
 
+    def sortby(self, variables, ascending=True):
+        """
+        Sorts the dataset, either along specified dimensions,
+        or according to values of 1-D dataarrays that share dimension
+        with calling object.  If multiple sorts along the same dimension is
+        given, numpy's lexsort is performed along that dimension:
+        https://docs.scipy.org/doc/numpy/reference/generated/numpy.lexsort.html
+        and the FIRST key in the sequence is used as the primary sort key,
+        followed by the 2nd key, etc.
+
+
+        Parameters
+        ----------
+        variables: (str, DataArray, or list of either)
+            Name(s) of a 1D variable in coords/data_vars whose values are used
+            to sort the dataset.
+        ascending: boolean, optional.
+            whether to sort by ascending order.
+
+        Returns
+        -------
+        sorted: Dataset
+            A new dataset where all the specified dims are sorted by dim
+            labels.
+        """
+        from .dataarray import DataArray
+
+        if not isinstance(variables, list):
+            vs = [variables]
+        else:
+            vs = variables
+        vs = [v if isinstance(v, DataArray) else self[v] for v in vs]
+
+        vars_by_dim = defaultdict(list)
+        for d in vs:
+            if len(d.dims) > 1:
+                raise ValueError("Input DataArray has more than 1 dimension.")
+            else:
+                key = d.dims[0]
+                if len(d) != self.dims[key]:
+                    raise ValueError("Input DataArray must have same "
+                                     "length as dimension it is sorting.")
+            vars_by_dim[key].append(d)
+
+        indices = {}
+        for key, ds in vars_by_dim.items():
```

Comment:
This is a minor point -- but can you use slightly longer variable names instead of the abbreviations for variables that exist outside of one line? e.g.,
`vs` -> `variables`
`d` -> `data_array`
`ds` -> arrays (this one is especially confusing, because `ds` is normally used for `xarray.Dataset`)

This would help for readability (also follows PEP8 guidelines)